### PR TITLE
Prefer full async solution to URL.

### LINKED ISSUE #1: Implement Async/ Await for URL
Right now o.url only supports completion blocks. Also it only supports GET and POST. We want to be able to use Swift's async/ await features. We will want to keep the completion block functions if async/ await isn't supported in some cases.

### PR DESCRIPTION
**Pull request by 0xLeif**
Closes #1 

Decided to remove the completion handlers and prefer only async.

### CODE REVIEW

File: Sources/o/URL/o+url.swift [hunk 1]
```diff
@@ -9,97 +9,181 @@ import Foundation
...
+        headerFields: [String: String]
+    ) async throws -> URLResponse? {
+        try await URLSession.shared.data(
+            for: request(
+                for: url,
+                method: .HEAD,
+                headerFields: headerFields
             )
-            .resume()
+        ).1
```
Comment (haIIux) [inline Sources/o/URL/o+url.swift:68]: Wwhat are the `.1`'s? I see them at the end of most of these functions.

**Review by haIIux — APPROVED** [review]
Only one question I left regarding the `.1` use.

File: Sources/o/URL/o+url.swift (hunk 1 above)
Comment (0xLeif) [reply Sources/o/URL/o+url.swift:68]: URLSession data function returns a tuple, `(Data?, URLResponse?)` and `.0` is Data and `.1` is the Response.

Comment (0xLeif) [reply Sources/o/URL/o+url.swift:68]: Some HTTP method shouldn't return Data I believe. So those ones have the `.1` to only return the response.

Comment (haIIux) [reply Sources/o/URL/o+url.swift:68]: Understood!

**Pull request merged**